presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -575,18 +605,27 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
                             entry.getValue(),
                             builder.build(),
                             Optional.empty(),
-                            ImmutableList.of()));
+                            aggregations.values().stream()
+                                    .map(Aggregation::getOrderBy)
+                                    .flatMap(Collection::stream)
+                                    .collect(toImmutableList())));
         }
 
         AggregationNode aggregationNode = new AggregationNode(
                 idAllocator.getNextId(),
                 subPlan.getRoot(),
-                aggregations.build(),
+                aggregations,
                 groupingSymbols,
                 AggregationNode.Step.SINGLE,
                 Optional.empty(),
                 groupIdSymbol);
 
+        if (aggregationNode.hasEmptyGroupingSet() && aggregationNode.hasNonEmptyGroupingSet() && !aggregationNode.getOrderBySymbols().isEmpty()) {
+            throw SemanticExceptions.notSupportedException(
+                    node,
+                    "ORDER BY in aggregate function with empty grouping set");
```

Comment:
It's harder during analysis since aggregation functions and group bys are at different parts of the query. This is sort of a natural place they come together. If you have suggestions on where to put it, I'm happy to make the change.